v19.1/bytes.md

```diff
@@ -66,6 +66,41 @@ valid UTF-8 byte sequences; an error is reported otherwise.
 `STRING` values can be cast explicitly to `BYTES`. This conversion
 always succeeds.
 
+### `STRING` vs. `BYTES`
+
+While both `STRING` and `BYTES` can appear to have similar behavior in many situations, one should understand their nuance before casting one into the other.
+
+`STRING` treats all of its data as characters, or more specificially, unicode code points. `BYTES` treats all of its data as a byte string. This difference in implementation can lead to dramatically different behavior. For example, let's take a complex ASCII character such as ☃ ([the snowman emoji](https://emojipedia.org/snowman/)):
+
+{% include copy-clipboard.html %}
+~~~ sql
+SELECT length('☃'::string);
+~~~
+
+~~~
+  length
++--------+
+       1
+~~~
+
+~~~ sql
+SELECT length('☃'::bytes);
+~~~
+~~~
+  length
++--------+
+       3
+~~~
+
+In this case, `LENGTH(string)` measures the number of unicode code points present in the string, where as `LENGTH(bytes)` measures the number of bytes required to store that value. Each character (or unicode code point) can be encoded using multiple bytes, hence the difference in output between the two.
```

Comment:
- Suggest linking to `LENGTH` built-in if possible.
- We should capitalize "Unicode" wherever it's used, as it's a proper noun.
- `s/where as/whereas/`
- We could link to more information about [code points](https://en.wikipedia.org/wiki/Code_point) but IDK if it's really necessary.